mesonbuild/interpreter.py

```diff
@@ -3972,9 +3978,21 @@ def add_arguments(self, node, argsdict, args_frozen, args, kwargs):
             argsdict[lang] = argsdict.get(lang, []) + args
 
     @noKwargs
-    @noPosargs
     def func_environment(self, node, args, kwargs):
-        return EnvironmentVariablesHolder()
+        if args:
+            FeatureNew('environment positional arguments', '0.52.0').use(self.subproject)
+        if len(args) > 1:
+            raise InterpreterException('environment takes only one optional positional arguments')
+        elif len(args) == 1:
+            initial_values = args[0]
+            if not isinstance(initial_values, dict):
+                raise InterpreterException('environment first argument must be a dictionary')
+        else:
+            initial_values = {}
+        env = EnvironmentVariablesHolder()
+        for k, v in initial_values.items():
+            env.set_method([k, v], {})
```

Comment:
While at it... what about also supporting this syntax too:
```meson
environment({'FOO' : ['BAR', 'BAZ']}, separator : ';')
```

maybe it's not worth it, just throwing the idea.

Reply:
I don't quite understand what that would do, I'll leave it for now.

Reply:
Forget that, it's not worth it :-)